api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -144,4 +145,33 @@
      * @return the configuration sources
      */
     Iterable<ConfigSource> getConfigSources();
+
+    /**
+     * Convert an abritrary value to the specified type using the converters known to this instance
+     * 
+     * @param <T>
+     *             the conversion type
+     * @param value
+     *             the value to convert
+     * @param type
+     *             the type into which the value should be converted
+     * @return the converted value
+     * @throws java.lang.IllegalArgumentException if the value cannot be converted to the specified type.
+     */
+    default <T> T convert(String value, Class<T> type) {        
+        return value == null ? null
```

Comment:
I think we shouldn't return null directly if the value is null. How about if I want to have a converter that picks up nulls and convert them to a value of my choice?

Reply:
Converters should only return `null` if the string reflected an empty value.  Converters should never be passed `null` as an argument.

Reply:
In that case, shouldn't we adjust the javadoc of org.eclipse.microprofile.config.spi.Converter#convert and advise to throw an IllegalArgumentException if the value parameter is null?

Reply:
We could do so.

Reply:
Per our conversation several weeks back, I'm inclined to remove this method. The addition of `getConverter` can solve this use case and more elegantly handles the scenario where no suitable converter is available. It would also confine null handling to the converter.

Reply:
Sure go ahead and remove it; also could you please squash the commits into one?

Reply:
Agree. Remove and clarify javadoc in org.eclipse.microprofile.config.spi.Converter#convert.

Reply:
The PR/commit title should be updated accordingly as well.

Reply:
As far as disallowing `null` as an argument to `Converter::convert`, do we prefer `NullPointerException` or `IllegalArgumentException`? I think NPE is more conventional - e.g. `Objects::requireNonNull` - even if the latter is more precise.

Reply:
I spun up #539 so that decision doesn't get hidden in this PR. I think we should track it separately.